src/js/providers/flash.js

```diff
@@ -111,6 +111,27 @@ define([
             };
         }
 
+        function checkFlashBlocked() {
+            var _this = this;
```

Comment:
@pajong because you are calling `checkFlashBlocked` from `onFocus`, the context is `window` not the provider instance.

You need to test this feature in 3 modes. Currently the throttled case works, but it's broken when blocking the plugin outright:

![screen shot 2016-01-22 at 12 04 17 pm](https://cloud.githubusercontent.com/assets/333258/12517381/638bf504-c100-11e5-87dd-e281a244f9f4.png)